events/views.py

```diff
@@ -69,15 +70,14 @@ def query_participation(user, events):
             boys.append(get_profile(party, "m"))
             divers.append(get_profile(party, "d"))
             participants.append(party)
-            part = Participation.objects.get(event=e, person=user)
+            participation.append(Participation.objects.get(event=e, person=user))
         except Participation.DoesNotExist:
             Participation(event=e, person=user).save()
```

Comment:
but what if the `Participation.objects.get(...)` fails with the DoesNotExist error that you catch in line 74/75? Then participation will have less entries than e.g. participants.

I think the except-branch should also append to the participation-list.

Also  the try-block could be limited to the Participation.objects.get: https://github.com/flo-schu/diverssite/pull/54/files

```python
for event in events:
    ...
    try:
       part = Participation.objects.get(event=e, person=user)
    except Participation.DoesNotExist:
       part = Participation(event=e, person=user) # todo: does the database add some default values? do we need to refresh?
       part.save()
    participation.append(part)
```

Reply:
super! That is much clearer. I've tested it and I think it solves the problem. I'll try merging all changes now